src/Ampersand/PatchHelper/Helper/Magento2Instance.php

```diff
@@ -119,6 +102,119 @@ public function __construct(string $path)
             $libPath = sanitize_filepath($dirList->getRoot(), $libPath) . '/';
             $this->listOfPathsToLibrarys[$libPath] = $lib;
         }
+
+        $this->prepareThemes();
+    }
+
+    /**
+     * Prepare theme configuration and a list of paths to ignore in the scanning for hyva projects
+     *
+     * @link https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/issues/75
+     *
+     * As I understand it the reqs are
+     * - Only include frontend themes that have a Hyva/ root
+     * - We can just ignore file changes to theme files from non Hyva themes, don't need to check them
+     *
+     * @TODO hyva_theme_fallback/general/theme_full_path may require re-adding back in some excluded themes
+     *
+     * @return void
+     */
+    private function prepareThemes()
+    {
+        $themeList = $this->objectManager->get(ThemeList::class);
+        $allFrontendThemes = $themesToIgnore = [];
+        foreach ($themeList as $theme) {
+            switch ($theme->getArea()) {
+                case Area::AREA_FRONTEND:
+                    $allFrontendThemes[$theme->getCode()] = $theme;
+                    break;
+                case Area::AREA_ADMINHTML:
+                    $this->customAdminThemes[] = $theme;
+                    break;
+            }
+        }
+        unset($theme);
+
+        $frontendThemIdToCode = [];
+        foreach ($allFrontendThemes as $frontendTheme) {
+            $frontendThemIdToCode[$frontendTheme->getId()] = $frontendTheme->getCode();
+        }
+
+        try {
+            /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
+            $storeManager = $this->objectManager->get(\Magento\Store\Model\StoreManagerInterface::class);
+            /** @var \Magento\Theme\Model\View\Design $design */
+            $design = $this->objectManager->get(\Magento\Theme\Model\View\Design::class);
+
+            $usedFrontendThemes = [];
+            foreach ($storeManager->getStores(false) as $store) {
+                $themeCode = $design->getConfigurationDesignTheme(
+                    'frontend',
+                    [
+                        'store' => $store->getId()
+                    ]
+                );
+
+                /**
+                 * @link https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Theme/Model/Theme/Resolver.php#L57-L63
+                 */
+                if (is_numeric($themeCode)) {
+                    $themeCode = $frontendThemIdToCode[$themeCode];
```

Comment:
I tried to execute that (merged with master) and got:

```
PHP Warning:  Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162

Warning: Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162
PHP Warning:  Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162

Warning: Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162
PHP Warning:  Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162

Warning: Undefined array key 7 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162
PHP Warning:  Undefined array key 9 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162

Warning: Undefined array key 9 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162
PHP Warning:  Undefined array key 9 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162

Warning: Undefined array key 9 in /projects/tools/ampersand-magento2-upgrade-patch-helper/src/Ampersand/PatchHelper/Helper/Magento2Instance.php on line 162
```

Reply:
Thanks @amenk yeah i need to work out this issue

https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/pull/80#discussion_r1031259204

to handle themes stored by ID and not code

I'm thinking I can probably handle it like the core does here https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Theme/Model/Theme/Resolver.php#L57-L63

Reply:
Cool - I think that should work. Currently the ID in the array is just empty. Do you have time for this or shall I try? Currently in the middle of a Hyvä upgrade ... but we can also first look at the Plugins :-)

Reply:
This already removes the warnings:

```
diff --git a/src/Ampersand/PatchHelper/Helper/Magento2Instance.php b/src/Ampersand/PatchHelper/Helper/Magento2Instance.php
index c310c3a..7ca22af 100644
--- a/src/Ampersand/PatchHelper/Helper/Magento2Instance.php
+++ b/src/Ampersand/PatchHelper/Helper/Magento2Instance.php
@@ -140,6 +140,8 @@ class Magento2Instance
             $frontendThemIdToCode[$frontendTheme->getId()] = $frontendTheme->getCode();
         }
 
+        $themeCollection = $this->objectManager->get(\Magento\Theme\Model\ResourceModel\Theme\Collection::class);
+
         try {
             /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
             $storeManager = $this->objectManager->get(\Magento\Store\Model\StoreManagerInterface::class);
@@ -159,7 +161,7 @@ class Magento2Instance
                  * @link https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Theme/Model/Theme/Resolver.php#L57-L63
                  */
                 if (is_numeric($themeCode)) {
-                    $themeCode = $frontendThemIdToCode[$themeCode];
+                    $themeCode = $themeCollection->getItemById($themeCode)->getCode();
                 }
 
                 if (isset($allFrontendThemes[$themeCode])) {
```

Reply:
I'll pop that in the branch later you hero

Reply:
@amenk , awesome, thanks a lot. Just wanted to start with this approach and saw, that you already fixed it. Cool.
I also have an open Hyva update, so will be happy to test further, as soon as this is merged.

Reply:
Pushed here @norgeindian @amenk 

https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/commit/f20afca1ad26143561979082715bd83793a41f0c